### DISCUSSION TITLE
October ESMValTool Workshop

### DISCUSSION THREAD

**Discussion by mo-tgeddes** [ESMValGroup/Community#45] [Meetings]
Hi,

The dates have been chosen for the workshop: October 18th - 20th (Tuesday - Thursday). You can attend either remotely or in person at the Met Office in Exeter, UK. In addition to the main workshop, two optional additional days for those who would like to work with the Iris developers are available on Monday 17th and Friday 21st.

Registration for in-person attendance is now closed, but remains open for online attendance:  https://forms.office.com/r/8L2C3nvWwx. 

Thanks,
Theo & Alistair

UPD: 
Programme for the workshop: https://docs.google.com/document/d/1iIBZnH0lV04XIqJC1XAviv5Fm5hO3MJkF44kR-nUMMw/edit?usp=sharing

**Comment by mo-tgeddes** [comment]
Hi @ESMValGroup/esmvaltool-developmentteam,
Could you please put any discussion points and session items you want to have at the workshop here please?
Thanks,
Theo.

**Reply by catherinehardacre** [reply]
Discussion point for the Workshop:
I'd really like to have a discussion about using station based observational data in ESMValTool. I'm currently trying to develop a dataset based on observations of Aerosol Optical Depth from the AeroNet network, but there is little guidance on how to proceed! If I can get an AeroNET AOD dataset developed for ESMVaTool (i.e. CMORized) Ranjini Swaminathan has expressed interest in including this data sets as an example/chapter in the ESMValTool tutorial.

**Reply by LisaBock** [reply]
We, the User Engagement Team, would like to have our regular team meeting during these three workshop days. Would it be possible to block for us one hour in the agenda and maybe provide an extra room? Thanks!

**Reply by zklaus** [reply]
In the shared Iris devs/ESMValTool core devs meeting we came up with the idea of having a "First experiences with iris-esmf-regrid" session.

**Reply by zklaus** [reply]
In discussion with @philipkershaw, we had the idea to have a presentation about the new search interface for ESGF that is based on STAC. Perhaps @agstephens could give it :smirk: ?

**Reply by agstephens** [reply]
Hi @zklaus, yes I could give a presentation on the emerging STAC work for ESGF. I have some commitments on those dates so please let me know possible agenda times soon so that I can confirm.

**Reply by alistairsellar** [reply]
Hi Ag, thanks for the offer! Agenda still under development so we can probably work around your availability. Mornings would be best - you have any availability in the mornings of 18, 19, 20 Oct?

**Reply by agstephens** [reply]
Thanks @alistairsellar, here is my availability:
18th: 0900-1100
19th: 0900-1100
20th: 0900-1200

**Reply by alistairsellar** [reply]
Thanks @agstephens. Can we put you down for 0900-1000 on the 19th then?

**Reply by agstephens** [reply]
Hi @alistairsellar @zklaus : yes, that is fine. How long do you want me to talk for?

**Reply by bouweandela** [reply]
There should be no need to talk for an hour if feel that you can tell the story in 15 minutes, but it's nice if we have time for questions and discussion, so I think it's good to book the full hour and then finish early if we're done early.

**Comment by alistairsellar** [comment]
Hi @ESMValGroup/esmvaltool-developmentteam here is some travel and accommodation information for those travelling to Exeter for the workshop...

**Accommodation**
There are two chain hotels within a 5-minute walk of the Met Office:
- Premier Inn Exeter (M5 J29): https://www.tripadvisor.co.uk/Hotel_Review-g186254-d5555181-Reviews-Premier_Inn_Exeter_M5_J29_hotel-Exeter_Devon_England.html
- Holiday Inn Express Exeter M5 JCT. 29: https://www.tripadvisor.co.uk/Hotel_Review-g186254-d189356-Reviews-Holiday_Inn_Express_Exeter_M5_JCT_29_an_IHG_Hotel-Exeter_Devon_England.html

There are several evening food options near to these, including a couple of pubs, a Dominoes Pizza and Subway. However, the Met Office and these hotels are 2.5 miles from the city centre and this edge-of-town area has nothing "exciting" in the way of nightlife, so those who prefer to be near the nightlife would be better to choose one of the many hotels in the city centre. We don't have any particular recommendations for these.

**Travel**
For those arriving by Eurostar, the simplest connection is to take the tube to London Paddington and on to Exeter St Davids. The train from Paddington to Exeter takes approx 2hr15 through rolling countryside (once you get past Reading). See this page for bus connections across Exeter to the Met Office: https://www.metoffice.gov.uk/about-us/contact/in-exeter (the number 56 goes direct from St Davids station to the Met Office).

For those arriving by air, 
- the nearest international airport is Bristol, and there are coaches direct from Bristol Airport to Exeter: https://www.stagecoachbus.com/promos-and-offers/south-west/south-west-falcon
- there are also direct coaches from London Heathrow to Exeter, operated by National Express: https://www.nationalexpress.com/en
- the Bristol and Heathrow coaches both stop at Honiton Road Park & Ride close to the Met Office and a 2-minute walk from the above hotels
- both airports also have connections to the rail network, for trains to Exeter St Davids: the [Bristol Flyer bus](https://flyer.bristolairport.co.uk/) takes approx 30 mins to Bristol Temple Meads; the [Heathrow Express train](https://www.heathrowexpress.com/#/) takes 15-20 mins from terminals 2, 3, 5 to London Paddington.

See https://www.metoffice.gov.uk/about-us/contact/in-exeter for additional travel information, and do ask us if you have any questions.

**Comment by alistairsellar** [comment]
**ESMValTool Workshop registration**

Hi all, the header of this discussion thread has been updated to link to a registration form for the ESMValTool workshop on 18-20 October. Please see https://github.com/ESMValGroup/Community/discussions/45 for details. The registration deadline is 26th September.

Please note that even if you filled in the availability poll in July, we still need you to register to confirm your attendance and provide some details to support the workshop organisation.

Thanks,
Alistair

**Reply by schlunma** [reply]
Could you maybe add some details about this UK Security Clearance? Is this something you need to apply for before entering the UK? Googling this was not really helpful :grimacing:

**Reply by alistairsellar** [reply]
No, is absolutely not needed!

But some of the UK attendees may already have it, and it helps our reception people if they know who does. Sorry for any confusion - I'll add a clarification to the form now.

**Reply by alistairsellar** [reply]
Thanks to everyone who has registered. We have 26 attending in person (split 50-50 between Met Office and others), and 16 registered for online attendance. Very exciting!

Registration for online attendance remains open: https://forms.office.com/r/8L2C3nvWwx. 

Hoping to share the draft programme next week...

**Comment by malininae** [comment]
Is there a rough program available? I wouldn't mind joining remotely, but being on the North American West Coast, I'd need to understand if I'd be able/ if it makes sense to join at all.

**Reply by alistairsellar** [reply]
Hi Liza, thanks for you interest and sorry for the delay. The programme is still under development, but these example topics give a rough flavour:
- Reports from teams: User Engagement Team, Technical Lead Developer Team, Scientific Lead Developer Team
- Parallel(?) team meetings: user, tech, science
- Backward incompatibility
- Governance update and next steps
- Station based observations
- ESGF search interface based on STAC
- First experiences with iris-esmf-regrid

I hope that helps?

**Reply by malininae** [reply]
@alistairsellar Thanks! I have registered, but just FYI, I won't be participating very actively (time zones :( )!

**Reply by alistairsellar** [reply]
Great! Looking forward to meeting you. Appreciate that it doesn't work well for your time zone!

**Comment by alistairsellar** [comment]
I'm excited to share a draft programme for the workshop: https://docs.google.com/document/d/1iIBZnH0lV04XIqJC1XAviv5Fm5hO3MJkF44kR-nUMMw/edit?usp=sharing 

If your name is in here, please can you email me to confirm your willingness (or not) to take the role proposed? 

Please also reply here to raise any questions about the programme.

**Comment by catherinehardacre** [comment]
Hi Alistair,

In a role reversal, I've deputised Jane to lead the station based observations discussion. Given the time difference, I'm unlikely to be able to join from NZ, unfortunately.

Catherine

**Reply by alistairsellar** [reply]
Thanks Catherine and Jane!

**Comment by rswamina** [comment]
@alistairsellar , @ehogan  and @remi-kazeroni  - If it's not too late, I'd like to include a topic for discussion - the distribution or availability of observation data sets on JASMIN/Levante. Some questions for discussion : 

- Should we have the full set made available on JASMIN and Levante (right now, I am told JASMIN only has a subset)?
- What is the best way to store, in a distributed yet accessible way CMO-ized obs data sets?
- Potential issues with access restriction etc.
We talked a bit about this at the June ESMValTool meeting but perhaps we can revisit the topic summarized under Observations in the summary from our June workshop. I couldn't find the document online so didn't link it here.

**Reply by alistairsellar** [reply]
Hi @rswamina I think this is an important topic and will be particularly important for us at Met Office when we get to encouraging / helping our model development partners in Aus/NZ/Korea/... to adopt ESMValTool on their own systems. It feels to me like some of the issues, particularly the license aspects, might benefit from a working-group type approach as we did for the backward incompatibility. My memory of the last workshop discussion is that it would be quite hard to reach decisions in the workshop discussion format. What do you think? We could have a 30-min discussion session at the workshop to review the primary outstanding issues and see who might be up for joining a working group to analyse possible solutions and make some recommendations. It would need someone to introduce the session, e.g. by listing the key blockers / challenges. (As ever, open to other ideas.)

**Reply by rswamina** [reply]
@alistairsellar - Yes, that sounds like a good plan. We can initiate a discussion next week and have a working group to formulate policy for obs. 
@remi-kazeroni  - Would you be interested in summarizing the current state and any plans for the future with regards to obs? I am happy to help.

**Reply by remi-kazeroni** [reply]
Hi @rswamina, @alistairsellar, I'd be happy to contribute to a session on obs data at the upcoming workshop if you think that is needed. During the last workshop, various aspects of the obs data sharing problem were discussed: legal issues, technicalities, Tier classification, ... see summary of previous workshops on the [website](https://www.esmvaltool.org/meetings.html). I'm not sure the situation has changed much since then but we can discuss the particular case of JASMIN if that can ease development of ESMValTool there.
As you may recall, the limitation to Tier2 CMORiezd data on Jasmin is not only due to legal concerns but also lack of storage, @valeriupredoi could tell us more on that. FYI, the storage requirements are:

- Tier2: 500 GB
- Tier3 (except ERA5): 1.5 TB
- RAWOBS/Tier3/ERA5/v1/mon: 300 GB
- RAWOBS/Tier3/ERA5/v1/*hr: 4 TB.
But we may not have to sync ERA5 data to Jasmin, I think these are already available there (tbc by actual Jasmin users).

**Reply by alistairsellar** [reply]
Great, thanks @rswamina & @remi-kazeroni! I've added the session to the programme and include both your names as leads.

**Reply by bouweandela** [reply]
> What is the best way to store, in a distributed yet accessible way CMOR-ized obs data sets?

The ESGF. There is the obs4MIPs project for observations and the CREATE-IP project for reanalyses. It would be good to try and collaborate more. Recently @glpotter from the CREATE-IP project joined us, so this is a really great start. If we can demonstrate that it works well for the CREATE-IP project, the obs4MIPs project might also be interested in collaborating with us.

**Reply by hb326** [reply]
How fortunate that I am sitting now on the obs4MIPS panel.  ;)
Let's brainstorm at the workshop how to make use of my new "position".

**Comment by malininae** [comment]
For those who are far away but would like to know updates. Would it be possible to record at least some pre-lunch discussions? I'll try to join on Tuesday and maybe Thursday, but I'm not guaranteeing that I will stay awake :-)

**Reply by alistairsellar** [reply]
Hi @malininae sorry for the delay - have been looking into this. I suspect that my institute is more cautious than others in this regard, but it seems that because it's not purely online we would need to get all participants to sign a consent form. And the form looks pretty full-on: includes something like granting the British Crown the rights to your image(!). 😨 Aside from the consent issue, I'm also a bit torn on this one: definitely keen to reach out to those like yourself who can't attend much, but conscious of some evidence that recording can discourage less confident people from contributing to the discussion at the time.

So we've discussed a bit and perhaps could instead provide an automated transcript of the discussions? MS Teams has this capability and it wouldn't seem to imply the need for lots of forms. Would this work for you @malininae? Would any participants have concerns about this, particularly with regard to putting them off getting involved in discussions? Discussion in the room would not be attributed to individuals (because they would all come via one mic); Teams would attribute discussion from remote participants to whatever handle they have in the meeting, but we could redact names on request.

**Reply by bouweandela** [reply]
@ESMValGroup/esmvaltool-developmentteam Opinions anyone on whether or not workshop sessions should be recorded and in what form?

If I recall correctly, we did record (some of) the sessions of the previous workshop.

**Reply by alistairsellar** [reply]
Yes, but it's not the same "we" in this case. Because Met Office are hosting, it would be us doing the recording and hence have different process that we need to follow according to local interpretation of GDPR and other regulations.

**Reply by bouweandela** [reply]
Maybe we could consider sharing the recordings only with regostered workshop participants? That will hopefully take away any reluctance to speak out.

**Reply by alistairsellar** [reply]
Perhaps it would, it's hard to know. But it wouldn't solve the issue of Met Office needing everyone to sign consent forms. I don't know why the Met Office is more restrictive than other organisations on this, but at this late stage there's nothing I can do to challenge our process.

I'm hoping that a transcript provides almost as much value as a recording?

**Reply by malininae** [reply]
@alistairsellar @bouweandela Ok, got you! I work for the Canadian Government (same crown, lol), and we also have to have special security assessments and other stuff to record the meetings, so I know the drill. In that sense, if others are not so interested in the recordings, then maybe event transcripts are not needed. I'll brew a big pot of coffee :-)   And maybe as you say, will just reach out to presenters in case of questions.

**Reply by axel-lauer** [reply]
Back then, we only recorded the presentations (with the consent of the presenter). To me, that's a different story than recording discussions. I think automatic transcripts would be fine (but possibly also not extremely useful to everyone). In the past, we always provided workshop summaries with the most important points discussed. Maybe that would be enough?

**Reply by alistairsellar** [reply]
Hi @bouweandela, I realised after I switched off the laptop yesteday that my tone was a bit dismissive in my replies to you. Sorry about that. I think I was a bit nervous about opening up discussion on something I wouldn't know how to fix. But that's not the right spirit. So... all, I too would be interested in hearing thoughts on the merits or otherwise of recording presentations.

Thanks @malininae & @axel-lauer for your replies too. @malininae, the transcripts are quite straightforward to generate, so unless anyone has concerns I'm happy to switch them on even if they are only a little useful to you.

**Comment by alistairsellar** [comment]
📢  **In-person attendees** 📢 

I've just emailed some detailed information to in-person attendees. Please shout if you believe that you have registered to attend in-person but didn't get an email from me!

**Comment by alistairsellar** [comment]
Thanks everyone who participated in last week's ESMValTool workshop - it was a blast! 

We'll email registered participants soon with links to transcripts and presentations. In the meantime, we thought it'd be nice if participants could post here some things that were achieved, or progressed, or some next steps that they plan to follow up, e.g. sparked by conversations or plans that were made.

**Reply by alistairsellar** [reply]
I'll kick off with a couple of my own examples:
 - Backward compatibility policy finalised (thanks especially @ehogan @bouweandela @axel-lauer). Draft PR created to get this into documentation: ESMValGroup/ESMValTool#2879
 - Started work on ESMValGroup/ESMValTool#2309. Didn't progress it as much as hoped due to the huge number of really useful conversations & meetings, but hopefully now have the momentum to progress it for v2.8.

Next steps: complete ESMValGroup/ESMValTool#2309

**Reply by bouweandela** [reply]
Thanks for organizing! Here are my personal highlights:

- We had some good team-building sessions with the @ESMValGroup/technical-lead-development-team, thanks everyone!
- First steps towards lazy saving with iris in https://github.com/SciTools/iris/pull/5031 together with @pp-mo @bjlittle and @zklaus. This will hopefully lead to much lower memory consumption and improved parallelization. To try it out, have a look at https://github.com/ESMValGroup/ESMValCore/pull/1714. Currently, this saves via xarray and loses all kinds of metadata in the process so it's only useful for performance testing, but once https://github.com/SciTools/iris/pull/5031 is merged and released it can be used for real work.
- Support lazy vertical interpolation with Python stratify https://github.com/SciTools/python-stratify/pull/52. Now we're waiting on @bjlittle to make a new release and then we can merge https://github.com/ESMValGroup/ESMValCore/pull/1761 and have lazy vertical interpolation in ESMValCore!

**Reply by axel-lauer** [reply]
Other achievements accomplished during the workshop include:

* Finalized and merged our new broken recipe policy (https://github.com/ESMValGroup/ESMValTool/pull/2720)
* Finalized and merged a documentation extension on the role of a recipe maintainer (https://github.com/ESMValGroup/ESMValTool/pull/2719)

**Reply by nhsavage** [reply]
working with Iris developers two fixes important for regridding CORDEX data were completed:
 * Prioritise dim coord in _get_lon_lat_coords()  https://github.com/SciTools/iris/pull/5029 
 * Allow regridding of cubes with multiple coordinates per axis https://github.com/SciTools-incubator/iris-esmf-regrid/pull/217

**Reply by schlunma** [reply]
Thanks for organizing!!

- Resolved https://github.com/SciTools/iris/issues/4860 with https://github.com/SciTools/iris/pull/5020
- had very good discussion regarding UGRID with Iris developers

**Reply by remi-kazeroni** [reply]
Thanks for organising the workshop!

- Discussion on the distribution and availability of observational data. Consensus to establish a working-group to formulate the policy for obs data.
- Discussion with @bjlittle and @bsolino on support for Zarr in Iris for usage within ESMValTool
- Discussion with @egalytska and @bsolino on possible ways to connect a Causal Discovery Python package to ESMValTool.
